src/aiortc/codecs/vpx.py

```diff
@@ -436,6 +436,275 @@ def __update_config(self) -> None:
+        assert isinstance(frame, VideoFrame)
+        if frame.format.name != "yuv420p":
+            frame = frame.reformat(format="yuv420p")
+
+        if self.codec and (frame.width != self.cfg.g_w or frame.height != self.cfg.g_h):
+            lib.vpx_codec_destroy(self.codec)
+            self.codec = None
+
+        self.enable_gcc = enable_gcc
+        if not self.codec:
+            # create codec
+            self.codec = ffi.new("vpx_codec_ctx_t *")
+            self.cfg.g_timebase.num = 1
+            self.cfg.g_timebase.den = VIDEO_CLOCK_RATE
+            self.cfg.g_lag_in_frames = 0
+            self.cfg.g_threads = number_of_threads(
+                frame.width * frame.height, multiprocessing.cpu_count()
+            )
+            self.cfg.g_w = frame.width
+            self.cfg.g_h = frame.height
+            self.cfg.rc_resize_allowed = 0
+            self.cfg.rc_end_usage = lib.VPX_CBR
+            """ quantizer = -1 is the full range quantizer in range 0 and 63 """
+            self.cfg.rc_min_quantizer = quantizer if quantizer > 0 else 0
+            self.cfg.rc_max_quantizer = quantizer if quantizer > 0 else 63
+            self.cfg.rc_undershoot_pct = 100
+            self.cfg.rc_overshoot_pct = 15
+            self.cfg.rc_buf_initial_sz = 500
+            self.cfg.rc_buf_optimal_sz = 600
+            self.cfg.rc_buf_sz = 1000
+            self.cfg.kf_mode = lib.VPX_KF_AUTO
+            self.cfg.kf_max_dist = 3000
+            self.vpx_min_bitrate = MIN_BITRATE
+            self.vpx_max_bitrate = MAX_BITRATE
+            if quantizer < 0 and not enable_gcc and MIN_BITRATE != MAX_BITRATE:
+                """ Setting the bitrate to target_bitrate in quantizer when full range
+                    if MIN_BITRATE != MAX_BITRATE
+                """
+                vpx_bitrate = self.get_vpx_bitrate(target_bitrate)
+                if vpx_bitrate is not None:
+                    self.vpx_min_bitrate = vpx_bitrate
+                    self.vpx_max_bitrate = vpx_bitrate
+                    self.__target_bitrate = vpx_bitrate
+
+            logger.info(f"encoder's config: min_quantizer {str(self.cfg.rc_min_quantizer)}, "\
+                        f"max_quantizer {str(self.cfg.rc_max_quantizer)}, "\
+                        f"kf_max_dist {str(self.cfg.kf_max_dist)}, "\
+                        f"DEFAULT_BITRATE {self.__target_bitrate}, MIN_BITRATE {self.vpx_min_bitrate}, "\
+                        f"MAX_BITRATE {self.vpx_max_bitrate}, "\
+                        f"buf_optimal_sz {str(self.cfg.rc_buf_optimal_sz)}, buf_sz {str(self.cfg.rc_buf_sz)}, "\
+                        f"buf_initial_sz {str(self.cfg.rc_buf_initial_sz)}, ")
+            self.__update_config()
+            _vpx_assert(lib.vpx_codec_enc_init(self.codec, self.cx, self.cfg, 0))
+
+            lib.vpx_codec_control_(
+                self.codec, lib.VP9E_SET_NOISE_SENSITIVITY, ffi.cast("int", 1)
+            )
+
+            lib.vpx_codec_control_(
```

Comment:
I assume these variables are identical?

Reply:
Yes, some of the `VP8E_` variables are shared between VP8 and VP9.